waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -316,12 +320,15 @@ proc handleRemovedEvents(g: OnchainGroupManager, blockTable: BlockTable):
 
 proc getAndHandleEvents(g: OnchainGroupManager,
                         fromBlock: BlockNumber,
-                        toBlock: BlockNumber): Future[void] {.async: (raises: [Exception]).} =
+                        toBlock: BlockNumber): Future[bool] {.async: (raises: [Exception]).} =
```

Comment:
I think that this proc either returns true or raises an exception.

Am I missing something or is it supposed to be like that? If so, what's the point of returning `bool` if the return value is always true?

Reply:
that's for compatibility with the retryWrapper template, it expects a return value to assign the result of an async operation, just set to true here since we don't use it (discarded)